subworkflows/local/map.nf

```diff
@@ -128,6 +131,38 @@ workflow MAP {
         ch_versions        = ch_versions.mix ( CIRCULARMAPPER.out.versions )
         ch_mapped_lane_bam      = CIRCULARMAPPER.out.bam
         ch_mapped_lane_bai      = CIRCULARMAPPER.out.bai // [ [ meta ], bai/csi ]
+    } else if ( params.mapping_tool == 'mapad' ) {
+        ch_input_for_mapping = reads
+                            .combine( index.map{ meta, index, fasta -> [ meta, index ] } )
+                            .multiMap {
+                                meta, reads, meta2, index ->
+                                    new_meta = meta + [ reference: meta2.id ]
+                                    reads: [ new_meta, reads ]
+                                    index: [ meta2, index ]
+                                    strandedness: meta.strandedness=="double"
+                            }
+
+        MAPAD_MAP (
+                ch_input_for_mapping.reads,
+                ch_input_for_mapping.index,
+                params.mapping_mapad_p,
+                ch_input_for_mapping.strandedness,
+                params.mapping_mapad_f,
```

Comment:
Just to check, all of these parameters would make sense to apply across all samples/genomes in a run?